river/Config.zig

```diff
@@ -178,15 +178,24 @@ pub fn deinit(self: *Self) void {
     self.xkb_context.unref();
 }
 
+const FilterKind = enum {
+    @"app-id",
+    title,
+};
+const FilterEffect = enum {
+    float,
+    csd,
+};
+
 pub fn shouldFloat(self: Self, view: *View) bool {
     if (view.getAppId()) |app_id| {
-        if (self.float_filter_app_ids.contains(std.mem.span(app_id))) {
+        if (self.matchesFilterPattern(FilterKind.@"app-id", FilterEffect.float, std.mem.span(app_id))) {
```

Comment:
```suggestion
        if (self.matchesFilterPattern(.@"app-id", .float, std.mem.span(app_id))) {
```

Zig can infer the type here.